src/main/java/uk/gov/hmcts/reform/civil/model/docmosis/sealedclaim/SealedClaimResponseForm.java

```diff
@@ -0,0 +1,172 @@
+
+    public String getResponseTypeDisplay() {
+        return responseType.getDisplayedValue();
+    }
+
+    public static SealedClaimResponseForm toSealedClaimResponseCommonContent(CaseData caseData) {
+        SealedClaimResponseForm.SealedClaimResponseFormBuilder builder = SealedClaimResponseForm.builder();
+
+        if (caseData.getRespondent1ClaimResponseTypeForSpec() != null) {
+            builder.howToPay(caseData.getDefenceAdmitPartPaymentTimeRouteRequired());
+            switch (caseData.getRespondent1ClaimResponseTypeForSpec()) {
+                case FULL_ADMISSION -> addRepaymentMethod(caseData, builder, caseData.getTotalClaimAmount());
+                case PART_ADMISSION -> partAdmissionData(caseData, builder);
+                case FULL_DEFENCE -> fullDefenceData(caseData, builder);
+                case COUNTER_CLAIM -> builder.whyReject(COUNTER_CLAIM.name());
+                default -> builder.whyReject(null);
+            }
+        }
+
+        return builder
+            .whyNotPayImmediately(caseData.getResponseToClaimAdmitPartWhyNotPayLRspec())
+            .responseType(caseData.getRespondent1ClaimResponseTypeForSpec())
+            .mediation(caseData.getResponseClaimMediationSpecRequired() == YesOrNo.YES)
+            .build();
+    }
+
+    private static void addRepaymentMethod(CaseData caseData, SealedClaimResponseForm.SealedClaimResponseFormBuilder builder, BigDecimal totalAmount) {
+        if (caseData.isPayImmediately()) {
+            addPayByDatePayImmediately(builder, totalAmount);
+        } else if (caseData.isPayByInstallment()) {
+            addRepaymentPlan(caseData, builder, totalAmount);
+        } else if (caseData.isPayBySetDate()) {
+            addPayBySetDate(caseData, builder, totalAmount);
+        }
+    }
+
+    private static void addPayBySetDate(CaseData caseData, SealedClaimResponseForm.SealedClaimResponseFormBuilder builder, BigDecimal totalClaimAmount) {
+        builder.payBy(caseData.getRespondToClaimAdmitPartLRspec().getWhenWillThisAmountBePaid())
+            .amountToPay(totalClaimAmount + "")
+            .whyNotPayImmediately(caseData.getResponseToClaimAdmitPartWhyNotPayLRspec());
+    }
+
+    private static void addPayByDatePayImmediately(SealedClaimResponseForm.SealedClaimResponseFormBuilder builder, BigDecimal totalClaimAmount) {
+        builder.payBy(LocalDate.now()
+                          .plusDays(RespondentResponsePartAdmissionPaymentTimeLRspec.DAYS_TO_PAY_IMMEDIATELY))
+            .amountToPay(totalClaimAmount + "");
+    }
+
+    private static void addRepaymentPlan(CaseData caseData, SealedClaimResponseForm.SealedClaimResponseFormBuilder builder, BigDecimal totalClaimAmount) {
+        if (caseData.getRespondent1RepaymentPlan() != null) {
+            builder.repaymentPlan(RepaymentPlanTemplateData.builder()
+                                      .paymentFrequencyDisplay(caseData.getRespondent1RepaymentPlan().getPaymentFrequencyDisplay())
```

Comment:
_caseData.getRespondent1RepaymentPlan()_ DRY
Put in separate variable ;)

Reply:
I do not see the repetition except when we are calling getRespondent1RepaymentPlan() but then if we do put it in variable and then call it, in my opinion it will be still the same respondent1RepaymentPlan.getPayemtFrequency()  and responent1RepaymentPlan().getFristRepaymentDate()..